exchange/utils.go

```diff
@@ -998,20 +965,20 @@ func applyFPD(fpd map[openrtb_ext.BidderName]*firstpartydata.ResolvedFirstPartyD
 	}
 
 	if fpdToApply.Site != nil {
-		bidRequest.Site = fpdToApply.Site
+		reqWrapper.Site = fpdToApply.Site
 	}
 
 	if fpdToApply.App != nil {
-		bidRequest.App = fpdToApply.App
+		reqWrapper.App = fpdToApply.App
 	}
 
 	if fpdToApply.User != nil {
 		//BuyerUID is a value obtained between fpd extraction and fpd application.
 		//BuyerUID needs to be set back to fpd before applying this fpd to final bidder request
-		if bidRequest.User != nil && len(bidRequest.User.BuyerUID) > 0 {
-			fpdToApply.User.BuyerUID = bidRequest.User.BuyerUID
+		if reqWrapper.User != nil && len(reqWrapper.User.BuyerUID) > 0 {
+			fpdToApply.User.BuyerUID = reqWrapper.User.BuyerUID
 		}
-		bidRequest.User = fpdToApply.User
+		reqWrapper.User = fpdToApply.User
```

Comment:
Just checking, this statement will change the user object pointer on `reqWrapper` to point to the user object on `fpdToApply`. I believe the `fpdToApply` user object may contain changes to the user ext. Will the request wrapper behave correctly when `RebuildRequest` is called or will the ext changes be lost? I think the changes may be lost if a change is made to the `UserExt` object on the request wrapper that result in it being marked as dirty so that a user ext rebuild is triggered when `RebuildRequest` is called.
If you all agree that this is a problem, I think it could be fixed in a future PR after we have released ORTB 2.6 support since I think this issue has always existed.

Reply:
Yes, if `fpdToApply.User.ext` has been modified, then the wrapper will overwrite those changes when the request is rebuilt.

Reply:
Good point, we will need to fix it.